rangeChecker.Tests/RangePrintInCsvTests.cs

```diff
@@ -0,0 +1,50 @@
+using System.Collections.Generic;
+using Xunit;
+
+namespace Range.Tests
+{
+    public class RangePrintInCsvTests
+    {
+        void TestRange(List<int> values, string expectedOutput)
+        {
+            string outputFormat = "{0}-{1}, {2}\n";
+            string actualOutput = RangeChecker.GetContinuousRangesInCsv(values, outputFormat);
+            Assert.Equal(expectedOutput, actualOutput);
+        }
+
+        [Fact]
+        void TestRegularValues()
+        {
+            TestRange(new List<int>() { 3, 3, 5, 4, 10, 11, 12 }, "3-5, 4\n10-12, 3\n");
+            TestRange(new List<int>() { 3, 5, 4, 10, 14, 12, 3 }, "3-5, 4\n");
+            TestRange(new List<int>() { 3, 7, 5, 10, 11, 12, 10 }, "10-12, 4\n");
+        }
+
+        [Fact]
+        void TestContinuousValuesOnly()
+        {
+            TestRange(new List<int>() { 3, 4, 5, 6 }, "3-6, 4\n");
+            TestRange(new List<int>() { 3, 4, 4, 5, 3, 6 }, "3-6, 6\n");
+            TestRange(new List<int>() { 3, 3, 4, 5, 10, 11, 11, 12 }, "3-5, 4\n10-12, 4\n");

```

Comment:
Assumption is that single value doesn't contribute to range. But if a value is contributing to the count in the range for set >=2 then a single number should be considered too.
This could be an enhancement and not necessary since you have explicitly mentioned your assumption as a test case :+1: